base/linalg/tridiag.jl

```diff
@@ -559,15 +559,18 @@ function getindex{T}(A::Tridiagonal{T}, i::Integer, j::Integer)
 end
 
 function setindex!(A::Tridiagonal, x, i::Integer, j::Integer)
+    @boundscheck checkbounds(A, i, j)
     if i == j
-        A.d[i] = x
+        @inbounds A.d[i] = x
     elseif i - j == 1
-        A.dl[j] = x
+        @inbounds A.dl[j] = x
     elseif j - i == 1
-        A.du[i] = x
-    else
-        throw(ArgumentError("cannot set elements outside the sub, main, or super diagonals"))
+        @inbounds A.du[i] = x
+    elseif !iszero(x)
+        throw(ArgumentError(string("cannot set entry ($i, $j) off the ",
+            "main, sub, and super diagonals to a nonzero value ($x)")))
```

Comment:
If it's a scalar entry, then it's only going to be in one of the diagonals, not multiple. There's an argument for multiple ways of saying this, guess it's more what sounds better. The condition for it working is that it has to be in one of the 3, the condition that caused the error was that it wasn't in any of the 3.